android/app/src/main/java/org/uwaterloo/subletr/pages/chat/ChatListingPageView.kt

```diff
@@ -0,0 +1,160 @@
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import org.uwaterloo.subletr.R
+import org.uwaterloo.subletr.theme.subletrPink
+
+@Composable
+fun ChatListingPageView(
+	modifier: Modifier = Modifier,
+	viewModel : ChatListingPageViewModel = hiltViewModel()
+){
+	Column {
+		Box(
+			modifier = Modifier
+				.fillMaxWidth()
+				.height(100.dp)
+				.padding(10.dp)
+				.background(Color.Transparent),
+			contentAlignment = Alignment.BottomStart
+		) {
+			Text(
+				text = "Messages",
+				color = Color.Black,
+				fontSize = 40.sp,
+				fontWeight = FontWeight.Bold,
+				modifier = Modifier.padding(vertical = 8.dp)
+			)
+		}
+		displayEntries()
+
+	}
+
+}
+
+
+@Composable
+fun ListItem(name : String, last_msg : String) {
+	Surface(color = Color.LightGray,
+		modifier = Modifier
+			.padding(vertical = 5.dp, horizontal = 10.dp)
+			.clip(RoundedCornerShape(40.dp)),
+		) {
+
+
+
+		Column(modifier = Modifier
+			.fillMaxWidth()) {
+
+			Row(verticalAlignment = Alignment.CenterVertically) {
+				Icon(name)
+				Column(modifier = Modifier
+					.weight(1f)
+					.padding(vertical = 40.dp),
+					//horizontalAlignment = Alignment.CenterHorizontally,
+					verticalArrangement = Arrangement.Center,)
+
+				{
+					Text(text = "$name :",
+						fontSize = 20.sp,
+						fontWeight = FontWeight.Bold)
+					Text(text = last_msg,
+						fontSize = 15.sp,
+						color = Color.DarkGray)
+				}
+				notification()
+			}
+		}
+	}
+}
+
+@Composable
+fun Icon(name : String) {
+	LazyColumn(
+	) {
+		item {
+			Box(
+				modifier = Modifier
+					.padding(dimensionResource(id = R.dimen.l))
+					.clip(CircleShape),
+			) {
+				Box(
+					modifier = Modifier
+						.width(dimensionResource(id = R.dimen.xxl))
+						.height(dimensionResource(id = R.dimen.xxl))
+						.background(subletrPink),
+					contentAlignment = Alignment.Center
+				) {
+					Text(
+						text = name[0].toString(),
+						color = Color.White,
+						fontSize = 32.sp,
+						fontWeight = FontWeight.Bold
+					)
+				}
+
+			}
+		}
+	}
+}
+
+@Composable
+fun notification(n : Int = 1) {
+	LazyColumn(
+	) {
+		item {
+			Box(
+				modifier = Modifier
+					.padding(dimensionResource(id = R.dimen.s))
+					.clip(CircleShape),
+			) {
+				Box(
+					modifier = Modifier
+						.width(dimensionResource(id = R.dimen.l))
+						.height(dimensionResource(id = R.dimen.l))
+						.background(Color(0xFF6ABC95)),
+					contentAlignment = Alignment.Center
+				) {
+					Text(
+						text = n.toString(),
+						color = Color.White,
+						fontSize = 20.sp,
+						fontWeight = FontWeight.Bold
+					)
+				}
+
+			}
+		}
+	}
+}
+@Composable
+fun displayEntries(names : List<String> = List(2){"Alex Lin"},
+					msg : List<String> = List(2){"This is test dialog"}) {
+	assert(names.size == msg.size)
+	(names.indices).forEach { i ->
```

Comment:
Maybe just make tuples of names and messages in the `viewModel`?
We should try to reduce computations in the `view`.